symengine/polys/uintpoly_piranha.h

```diff
@@ -70,6 +108,36 @@ class UIntPolyPiranha : public UIntPolyBase<pintpoly, UIntPolyPiranha>
         return poly_.degree();
     }
 
+    // begin() and end() are unordered
+    // rbegin() and rend() are ordered
```

Comment:
Then we can't have the same function (eg. `print`) for all three classes. Two will use `rbegin` one will use `obegin`.